### PR TITLE
♻️ refactor: refactor the persist layer from zustand's persist  to dexie ORM

### LINKED ISSUE #493: [Bug] 主题搜索不搜索内容 描述
### 💻 系统环境 | Operating System

macOS

### 🌐 浏览器 | Browser

Safari

### 🐛 问题描述 | Bug Description

The global search function indexes all content from all conversation (see #1 in picture). However, the per-agent topic search indexes only the titles (see #3 in picture).

全局搜索功能索引了所有对话的所有内容（见图片中的#1）。然而，每个客服人员的主题搜索只索引标题（见图片中的#3）。

### 🚦 期望结果 | Expected Behavior

I would like the per-agent topic search to search through all content from each conversation.

我希望每个客服人员的主题搜索能够搜索每个对话的所有内容。

### 📷 复现步骤 | Recurrence Steps

1. Create an agent and ask it "Please list 6 citrus fruits".
2. Search in the main search list for "Grapefruit". Notice the agent you created.
3. Search in the agent's topic lists for "Grapefruit". Notice there are no results.

1. 创建一个客服人员，并询问它“请列出6种柑橘水果”。
2. 在主搜索列表中搜索“葡萄柚”。注意到你创建的客服人员。
3. 在客服人员的主题列表中搜索“葡萄柚”。注意到没有结果。

### 📝 补充信息 | Additional Information

Pasted_Image_11_21_23__09_40

### PR DESCRIPTION
**Pull request by arvinxx**
#### 💻 变更类型 | Change Type

- [x] ✨ feat
- [ ] 🐛 fix
- [x] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

- [x] 敲定新版数据结构，构造 session Model https://github.com/lobehub/lobe-chat/pull/500/commits/3eb48d74b87085a3138cb671de4e8a01f73cf89b
- [x] sessionStore 改造为 service 请求模式 https://github.com/lobehub/lobe-chat/pull/500/commits/bda0302d958ba9e41e5c689866f03edc70b0cd2d
- [x] 重构 chat  slice 为独立 store，并改造相应 service https://github.com/lobehub/lobe-chat/pull/500/commits/5915caa9e5507f70b11f4ef55e17540166f81ed3
- [x] 构造 messages model 与 service https://github.com/lobehub/lobe-chat/pull/500/commits/55e65cfaf81ecc895cc68fde13411da0a1ef567f
- [x] 构造 topic model 与 service https://github.com/lobehub/lobe-chat/pull/500/commits/3e7b05ebb8efb58092375546c7ccdaeb40d12f9c
- [x] 添加数据结构迁移方案 https://github.com/lobehub/lobe-chat/pull/500/commits/e6e0788ac5acb37c9033d76b39244ba10617642e
- [x] 迁移弹窗 https://github.com/lobehub/lobe-chat/pull/500/commits/57ef1e5036092d92444f3ac7cbbcb00b8babeb9d
- [x] 支持 session 的全文搜索 https://github.com/lobehub/lobe-chat/pull/500/commits/b40fb6a15d365ea612b4898f552f0eb3906cae73 
- [x] 支持 topic 的全文搜索  https://github.com/lobehub/lobe-chat/pull/500/commits/bb74a10a088a4d6a8eb1f4bf9de162c38a960958 ，close #493 

### 📝 补充信息 | Additional Information

refs: 
- #378 

#### 数据迁移、导入弹窗

1. 数据库迁移提示

  image

3. 数据导入弹窗

image

#### 性能测试

image

10w messages、200 topics 和 4 sessions 的联合查询，耗时 1s 左右：

image

补充说明：测试掉了很多，由于覆盖到了之前没覆盖的实现，在后续迭代中逐步补齐。

### CODE REVIEW

**Comment by Asuka109** [comment]
https://electric-sql.com/docs/integrations/drivers/web/wa-sqlite
Maybe we need to migrate the data persistence layer again if we switch to using electric.
How about just migrate to electric right now?

**Comment by arvinxx** [comment]
> https://electric-sql.com/docs/integrations/drivers/web/wa-sqlite
> 
> Maybe we need to migrate the data persistence layer again if we switch to using electric.
> 
> How about just migrate to electric right now?

I'm not sure to use electric-sql right now, due to it's a brand new library. We might meet some uncontrollable problem because of its immaturity and GPT may can't help us either.

By now dexie and yjs are more complete then electric-sql. On the way of the local-first, I prefer to implement the ability and ship great feature to users first, and then re-think which tool is better to reach the same result.

At this time before, I actually have little experience on indexedDB and dexie.But with GPT's help, I can also have a good sense of these library extremely easy. 
So I think we also need to consider of the AI-friendly level of the project when we try to introduce some libraries to our project.

File: src/app/chat/features/Conversation/ChatList/Messages/index.ts [hunk 1]
```diff
@@ -13,10 +13,10 @@ import { FunctionMessage } from './Function';
 import { UserMessage } from './User';
 
 export const renderMessages: ChatListProps['renderMessages'] = {
-  assistant: AssistantMessage,
-  default: DefaultMessage,
-  function: FunctionMessage,
-  user: UserMessage,
+  assistant: AssistantMessage as any,
```
Comment (arvinxx) [inline src/app/chat/features/Conversation/ChatList/Messages/index.ts:16]: 需要移除 any

File: src/app/chat/features/Conversation/ChatList/index.tsx [hunk 2]
```diff
@@ -59,15 +53,13 @@ const List = memo(() => {
 
   return (
     <ChatList
-      data={data}
+      data={data as any}
```
Comment (arvinxx) [inline src/app/chat/features/Conversation/ChatList/index.tsx:56]: ChatList 组件需要修改

**Pull request merged**